numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -634,6 +608,34 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self)
+{
+    if (PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
```

Comment:
Extra parens here

Reply:
No, I think we need them all... compiler errors for `if (PyDataType_HASFIELDS((PyVoidScalarObject*)self->descr))`

Reply:
My bad, I was incorrectly reading `if ((PyDataType_HASFIELDS(...)))` here